src/Control/Effect/NonDet.hs

```diff
@@ -37,6 +39,20 @@ data NonDet m k
 runNonDet :: (Alternative f, Applicative m) => NonDetC m a -> m (f a)
 runNonDet (NonDetC m) = m (fmap . (<|>) . pure) (pure empty)
 
+-- | Nondeterministically choose an element from a 'Foldable' collection.
+-- This can be used to emulate the style of nondeterminism associated with
+-- programming in the list monad:
+-- @
+--   pythagoreanTriples = do
+--     a <- choose [1..10]
+--     b <- choose [1..10]
+--     c <- choose [1..10]
+--     guard (a^2 + b^2 == c^2)
+--     pure (a, b, c)
+-- @
+choose :: (Foldable t, Alternative m) => t a -> m a
+choose = getAlt . foldMap (Alt . pure)
```

Comment:
What do you think about doing `foldMapA` instead? I personally find it much more generally useful, and it still enjoys good type inference properties.

Reply:
No objections at to exposing `foldMapA` as well as `choose`.

Reply:
At that point, given that `choose = foldMapA pure`, it seems like we don’t gain much from including both. Thoughts?

Reply:
I’m a big fan of providing a single verb for “choose an element of this structure”, for a few reasons: both because I find existing `foldMapA pure` or `asum . fmap pure` patterns unreadable, or at best non-obvious, and also because I think this operation is extremely common. (I’ve been practicing some simple logic programs recently, and I needed `choose` in all of them.)

Reply:
Ok, makes sense 👍

What do you think about renaming it to not conflict with #198’s smart constructor? Perhaps `oneOf`?

Reply:
`oneOf` is fine by me. `choose` was a little too close to `choice` from `parsers`. Will make the change!